From 767280fca85b56395d34f78a368172d115d1a50c Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Fri, 8 Jul 2016 13:01:57 -0500
Subject: [PATCH 11/21] Rewriting iga1_crtc_save

Rewrote iga1_crtc_save callback function, in order to improve ACPI S3
State resume reliability.

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 src/via_display.c | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/src/via_display.c b/src/via_display.c
index 939e205..b52554a 100644
--- a/src/via_display.c
+++ b/src/via_display.c
@@ -2811,20 +2811,12 @@ iga1_crtc_save(xf86CrtcPtr crtc)
     DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                         "Entered iga1_crtc_save.\n"));
 
-    vgaHWProtect(pScrn, TRUE);
-
-    /* Save the standard VGA registers. */
-    if (xf86IsPrimaryPci(pVia->PciInfo)) {
-        vgaHWSave(pScrn, &hwp->SavedReg, VGA_SR_ALL);
-    } else {
-        vgaHWSave(pScrn, &hwp->SavedReg, VGA_SR_MODE);
-    }
-
-    viaIGA1Save(pScrn);
-
     vgaHWProtect(pScrn, FALSE);
     vgaHWUnlock(hwp);
 
+    vgaHWSave(pScrn, &hwp->SavedReg, VGA_SR_ALL);
+    viaIGA1Save(pScrn);
+
     DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                         "Exiting iga1_crtc_save.\n"));
 }
-- 
1.9.1

